lib/kafo/kafo_configure.rb

```diff
@@ -375,7 +375,12 @@ def parse_app_arguments
       self.class.app_options.each do |option|
         name                    = option.attribute_name
         value                   = send(option.flag? ? "#{name}?" : name)
-        config.app[name.to_sym] = value.nil? ? option.default_value : value
+
+        if value.nil? && !option.default_value.nil?
```

Comment:
```suggestion
        if value.nil?
```
This would be closer to the previous behavior, yes?

If we do in fact want to add some new additional condition `&& !option.default_value.nil?`, shouldn't we also have some handling for when `value.nil? && option.default_value.nil?` such as a warning that the default value is undefined or an error that says this code path shouldn't have been traversed.

Reply:
I have updated this logic to hopefully make it clearer what is different here now and some test cases